frontend/scripts/actions/tool.actions.js

```diff
@@ -463,15 +463,13 @@ function getSelectedNodeIds(currentSelectedNodesIds, changedNodeId) {
 
 export function selectNode(nodeId, isAggregated = false) {
   return (dispatch, getState) => {
+    const { tool } = getState();
     if (isAggregated) {
       dispatch(selectView(true));
+    } else if (tool.areNodesExpanded) {
+      dispatch(selectExpandedNode(nodeId));
     } else {
-      const currentSelectedNodesIds = getState().tool.selectedNodesIds;
```

Comment:
Just tested this, this is a behavior change, and I'm not sure we have discussed it
- Start with a clean sankey
- Select "Bungee" and "Cargil" from exporter column
- Expand by those 2 nodes
- Select "Bungee" from importer column

In the previous code, this filters the flows without recomputing the sankey
With these changes, this recomputes the sankey

![image](https://user-images.githubusercontent.com/973964/33562864-ad3acbc8-d916-11e7-8e7e-d274529810de.png)

Reply:
Indeed, this is a change in behaviour. The thing is, that having the ability to select a node that isn't visible through search exposed and existing issue (previously limited to the map selection). Specifically what happened was the following:
1. Selected through the map a non visible municipality node.
2. Select an exporter.
3. Select a country.
If you deselected the original municipality node the result was all flows from that (now deselected municipality to the selected exporter/country). This to me makes no sense, the resulting flow should be all municipalities that pass through the selected exporter and through the selected country.
I showed this to @juancarlosalonso and we decided that expanded view should always recalculate on adding/removal of nodes.

Reply:
I see, and I agree we should change that behavior. However
- It seems to me that this is unrelated to this PR, so we should treat it in it's own PR
- We should review this more carefully because, as I pointed out above, this also affects other cases
- We should review these with Lena in our weekly meetings

Reply:
ok, can cherry pick this commits.